component/discovery/triton/triton.go

```diff
@@ -0,0 +1,79 @@
+package triton
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/config"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/prometheus/common/model"
+	prom_discovery "github.com/prometheus/prometheus/discovery/triton"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "discovery.triton",
+		Args:    Arguments{},
+		Exports: discovery.Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+type Arguments struct {
+	Account         string           `river:"account,attr"`
+	Role            string           `river:"role,attr,optional"`
+	DNSSuffix       string           `river:"dns_suffix,attr"`
+	Endpoint        string           `river:"endpoint,attr"`
+	Groups          []string         `river:"groups,attr,optional"`
+	Port            int              `river:"port,attr,optional"`
+	RefreshInterval time.Duration    `river:"refresh_interval,attr,optional"`
+	Version         int              `river:"version,attr,optional"`
+	TLSConfig       config.TLSConfig `river:"tls_config,block,optional"`
+}
+
+var DefaultArguments = Arguments{
+	Role:            "container",
+	Port:            9163,
+	RefreshInterval: 60 * time.Second,
+	Version:         1,
+}
+
+// SetToDefault implements river.Defaulter.
+func (args *Arguments) SetToDefault() {
+	*args = DefaultArguments
+}
+
+// Validate implements river.Validator.
+func (args *Arguments) Validate() error {
+	if args.RefreshInterval <= 0 {
+		return fmt.Errorf("refresh_interval must be greater than 0")
+	}
+
+	return nil
+}
+
+func (args *Arguments) Convert() *prom_discovery.SDConfig {
+	return &prom_discovery.SDConfig{
+		Account:         args.Account,
+		Role:            args.Role,
+		DNSSuffix:       args.DNSSuffix,
+		Endpoint:        args.Endpoint,
+		Groups:          args.Groups,
+		Port:            args.Port,
+		RefreshInterval: model.Duration(args.RefreshInterval),
+		TLSConfig:       *args.TLSConfig.Convert(),
+		Version:         args.Version,
+	}
+}
```

Comment:
I would suggest to always cover the Convert function with a test to make sure that the args are correctly passed to the prometheus discovery. Last time I assigned the TLS config without converting it and it compiled and ran without error. I only realised that the conversion was missing when I ran a test